R/splitModels.R

```diff
@@ -1,5 +1,16 @@
 splitModels <- function ( qMatrix = NULL , person.groups = NULL , split = c ( "qMatrix" , "person.groups" ) , add = NULL , cross = NULL , all.persons = TRUE , all.persons.lab = "all" , person.split.depth = 0:length(person.groups[,-1,drop=FALSE]), full.model.names = TRUE , model.name.elements = c ( "dim" , "group" , "cross" ) , include.var.name = FALSE , env = FALSE , nCores=NULL , mcPackage = c("future", "parallel"), GBcore=NULL , verbose = TRUE ) {
-    if(!is.null(qMatrix))       {qMatrix      <- checkQmatrixConsistency (qMatrix)}
+  # checks
+  lapply(c(all.persons, full.model.names, include.var.name, env, verbose), checkmate::assert_logical, len = 1)
+  lapply(c(split, model.name.elements), checkmate::assert_character, null.ok = TRUE)

```

Comment:
I would suggest a stricter test here, because only four values are allowed for `model.name.elements`. So maybe
```
checkmate::assert_character(split, null.ok = TRUE, max.len = 2, unique=TRUE)
checkmate::assert_character(model.name.elements, null.ok = FALSE, max.len = 4, unique=TRUE)
```